app/services/check_sys_admin_authorisation.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class CheckSysAdminAuthorisation
+  include ServicePattern
+
+  def initialize(session:)
+    @session = session
+  end
+
+  def call
+    signin_user =
+      if otp_feature_enabled?
+        OtpSignInUser.load_from_session(@session)
+      else
+        DfESignInUser.load_from_session(@session)
+      end
+
+    signin_user.present? && User.system_admins.kept.exists?(email: signin_user.email)
```

Comment:
can't we just ask `signin_user.system_admin?`?

Reply:
we could if we added that to `OtpSignInUser` and `DfESignInUser` but as things stand these classes are really just a thin wrapper around a couple of session variables.